openfl/experimental/runtime/local_runtime.py

```diff
@@ -62,6 +66,124 @@ def ray_call_get(self) -> List[Any]:
         return clones
 
 
+def ray_group_assign(collaborators, collaborators_per_group=3, max_concurrency=1):
+    # This function takes the list of collaborators and the collaborators per group and assigns the collaborators to resource groups which share cuda context,
+    # it returns a list of GroupMembers, that is a utility class that manages the collaborator and in which group it is in. This maintians compatablility with current runtime execution without having to manage the groups manually.
+    
+    class GroupMember:
+        # Utility class that maintains compatability with runtime execution,

```

Comment:
It's great you've added these comments. I would recommend converting these to docstrings so they're available in our upcoming revamped API documentation.

Reply:
Added docstrings